public/javascripts/Choropleth.js

```diff
@@ -1,241 +1,327 @@
+                var activity = params.webpageActivity+regionId+"_distanceLeft="+distanceLeft+"_target=audit";
+                postToWebpageActivity(activity);   
             });
         }
+        return regionData;
+    }
 
-        // adds the neighborhood polygons to the map
-        $.getJSON("/neighborhoods", function (data) {
-            neighborhoodPolygonLayer = L.geoJson(data, {
-                style: style,
-                onEachFeature: onEachNeighborhoodFeature
-            })
-                .addTo(map);
-        });
-
+    /**
+     * Takes a completion percentage, bins it, and returns the appropriate color for a choropleth.
+     *
+     * @param p {float} represents a completion percentage, between 0 and 100
+     * @returns {string} color in hex
+     */
+    function getColor(p) {
+        //since this is a float, we cannot directly compare. Using epsilon to avoid floating point errors
+        return Math.abs(p - 100) < Number.EPSILON ? '#03152f':
+                 p > 90 ? params.regionColors[0] :
+                    p > 80 ? params.regionColors[1] :
+                        p > 70 ? params.regionColors[2] :
+                            p > 60 ? params.regionColors[3] :
+                                p > 50 ? params.regionColors[4] :
+                                    p > 40 ? params.regionColors[5] :
+                                        p > 30 ? params.regionColors[6] :
+                                            p > 20 ? params.regionColors[7] :
+                                                p > 10 ? params.regionColors[8] :
+                                                    params.regionColors[9];
+    }
 
-        // Logs when a region is selected from the choropleth and 'Click here' is clicked
-        // Logs are of the form "Click_module=Choropleth_regionId=<regionId>_distanceLeft=<"0", "<1", "1" or ">1">_target=audit"
-        // Log is stored in WebpageActivityTable
-        $("#choropleth").on('click', '.region-selection-trigger', function () {
-            var regionId = $(this).attr('regionId');
-            var ratesEl = rates.find(function(x){
-                return regionId == x.region_id;
-            });
-            var compRate = Math.round(100.0 * ratesEl.rate);
-            var milesLeft = Math.round(0.000621371 * (ratesEl.total_distance_m - ratesEl.completed_distance_m));
-            var distanceLeft = "";
-            if (compRate === 100){
-                distanceLeft = "0";
-            }
-            else if (milesLeft === 0){
-                distanceLeft = "<1";
-            }
-            else if (milesLeft === 1){
-                distanceLeft = "1";
-            }
-            else{
-                distanceLeft = ">1";
+    /**
+     * This function finds the color for a specific region of the accessibility choropleth.
+     * 
+     * @param {*} rate Object from which information about labels is retrieved.
+     */
+    function findAccessibilityChoroplethColor(rate) {
```

Comment:
The `rate` parameter should probably be renamed. I don't think that fully captures what this object contains.